src/abstractdataframe/selection.jl

```diff
@@ -287,6 +293,11 @@ function normalize_selection(idx::AbstractIndex,
                              @nospecialize(sel::Pair{<:ColumnIndex, <:Base.Callable}), renamecols::Bool)
     c = idx[first(sel)]
     fun = last(sel)
+
+    if fun === AsTable
+        return normalize_selection(idx, first(sel) => identity => AsTable, renamecols)
```

Comment:
This looks good. No reason to worry about copies because it's only a vector of named tuples, so we always allocate.

Reply:
yes - `AsTable` as destination will always allocate:
```
julia> df = DataFrame(a=1)
1×1 DataFrame
 Row │ a
     │ Int64
─────┼───────
   1 │     1

julia> df2 = select(df, :a => identity => AsTable, copycols=false)
1×1 DataFrame
 Row │ x1
     │ Int64
─────┼───────
   1 │     1

julia> df2.x1 === df.a
false
```